packages/ra-core/src/sideEffect/fetch.js

```diff
@@ -38,6 +38,17 @@ function validateResponseFormat(
         );
         throw new Error('ra.notification.data_provider_error');
     }
+    if (
+        fetchActionsWithArrayOfRecordsResponse.includes(type) &&
```

Comment:
this will make all existing providers fail for the DELETE_MANY and UPDATE_MANY verbs, whose response format was left fuzzy (array of records or array of ids?), probably because it's not used.

Reply:
Should I limit this message to the `GET_LIST`, `GET_MANY` and `GET_MATCHING` then ?